In [ ]:
2 +2 


4

In [ ]:
5*7

35

In [ ]:
3**5

243

In [ ]:
10**0.5


3.1622776601683795

*Recordar:* $\sqrt[n]{a} = a^{1/n}$. Luego
$$
\sqrt{10} = 10^{1/2} = 10^{0.5}.
$$

In [ ]:
import math

In [ ]:
math.log(10)

2.302585092994046

In [ ]:
math.log10(10)





In [ ]:
math.sin(0)

0.0

In [ ]:
math.sin(360)

0.9589157234143065

In [ ]:
math.pi

3.141592653589793

In [ ]:
math.sin(2 * math.pi)

-2.4492935982947064e-16

La expresión `-2.4492935982947064e-16` se lee $-2.4492935982947064 \times 10^{-16}$. Es decir es
$$
-0.00000000000000024492935982947064
$$

Para devolver todas las funciones del módulo ejecutamos `dir(math)`:

In [ ]:
dir(math)

['__doc__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 'acos',
 'acosh',
 'asin',
 'asinh',
 'atan',
 'atan2',
 'atanh',
 'ceil',
 'copysign',
 'cos',
 'cosh',
 'degrees',
 'e',
 'erf',
 'erfc',
 'exp',
 'expm1',
 'fabs',
 'factorial',
 'floor',
 'fmod',
 'frexp',
 'fsum',
 'gamma',
 'gcd',
 'hypot',
 'inf',
 'isclose',
 'isfinite',
 'isinf',
 'isnan',
 'ldexp',
 'lgamma',
 'log',
 'log10',
 'log1p',
 'log2',
 'modf',
 'nan',
 'pi',
 'pow',
 'radians',
 'remainder',
 'sin',
 'sinh',
 'sqrt',
 'tan',
 'tanh',
 'tau',
 'trunc']

In [ ]:
math.degrees(math.pi)

180.0

In [ ]:
math.sin(math.radians(360))

-2.4492935982947064e-16

In [ ]:
math.e
math.log(math.e)

1.0

Recordemos que $\log(a) = b$ si y solo si $e^b = a$.

En  general 
$$
\log_a(x) = y \Leftrightarrow a^y = x. 
$$

La función `log1p(x)` devuelve `log(1 + x)`: 

In [ ]:
math.log1p(10) # 

2.3978952727983707

In [ ]:
math.log1p(math.e - 1)

1.0

In [ ]:
math.log(11)

2.3978952727983707

In [ ]:
math.pow(2, 3)

8.0

In [ ]:
2**3

8

In [ ]:
5 + math.inf / 2

inf

Para evitar escribir `math.` antes de cada función ejecutamos la siguiente celda  de código:

In [ ]:
from math import *
sin(2 * pi)

-2.4492935982947064e-16

Si sabemos que vamos a utilizar algunas funciones determinadas del módulo, podemos importar solamente esa funciones:

In [ ]:
from math import log, e
log(e)

1.0

Si queremos acortar o cambiar el nombre del módulo dentro de nuestro programa podemos hacer algo como lo siguiente:

In [ ]:
import math as mt
mt.log(mt.e)

1.0